src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -27,3 +29,120 @@ def show_version(cmd):  # pylint: disable=unused-argument
     from azure.cli.core.util import get_az_version_json
     versions = get_az_version_json()
     return versions
+
+
+def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many-locals, too-many-statements, too-many-branches, no-member, unused-argument
+    import os
+    import platform
+    import sys
+    import subprocess
+    import azure.cli.core.telemetry as telemetry
+    from azure.cli.core import __version__ as local_version
+    from azure.cli.core._environment import _ENV_AZ_INSTALLER
+    from azure.cli.core.extension import get_extensions, WheelExtension
+    from distutils.version import LooseVersion
+    from knack.util import CLIError
+
+    update_cli = True
+    from azure.cli.core.util import get_latest_from_github
+    try:
+        latest_version = get_latest_from_github()
+        if latest_version and LooseVersion(latest_version) <= LooseVersion(local_version):
+            logger.warning("You already have the latest azure-cli version: %s", local_version)
+            update_cli = False
+            if not update_all:
+                return
+    except Exception as ex:  # pylint: disable=broad-except
+        logger.debug("Failed to get the latest version. %s", str(ex))
+    exts = [ext.name for ext in get_extensions(ext_type=WheelExtension)] if update_all else []
+
+    exit_code = 0
+    installer = os.getenv(_ENV_AZ_INSTALLER) or ''
+    installer = installer.upper()
+    if update_cli:
+        latest_version_msg = 'It will be updated to {}.'.format(latest_version) if yes \
+            else 'Latest version available is {}.'.format(latest_version)
+        logger.warning("Your current Azure CLI version is %s. %s", local_version, latest_version_msg)
+        from knack.prompting import prompt_y_n
+        if not yes:
+            confirmation = prompt_y_n("Please check the release notes first: https://docs.microsoft.com/"
+                                      "cli/azure/release-notes-azure-cli\nDo you want to continue?", default='y')
+            if not confirmation:
+                telemetry.set_success("Upgrade stopped by user")
+                return
+
+        if installer == 'DEB':
+            from azure.cli.core.util import in_cloud_console
+            if in_cloud_console():
+                raise CLIError("az upgrade is not supported in Cloud Shell.")
+            apt_update_cmd = 'apt-get update'.split()
+            az_update_cmd = 'apt-get install --only-upgrade -y azure-cli'.split()
+            if os.geteuid() != 0:  # pylint: disable=no-member
+                apt_update_cmd.insert(0, 'sudo')
+                az_update_cmd.insert(0, 'sudo')
```

Comment:
What happens if the current user does not have sudo permission?

Reply:
An error will be thrown by the Linux system and shown to the user.

Reply:
If the error message is given by the OS, and the message does not confuse the user, then I think it is fine~
It may be better that let the users input their admin password to obtain sudo permissions.